patches/7750.diff

```diff
@@ -0,0 +1,293 @@
+         LOG.log(Level.FINER, "didOpen: {0}", params);
+diff --git a/java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImplTest.java b/java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImplTest.java
+index 0f2bda50ae..06fd93d3e5 100644
+--- a/java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImplTest.java
++++ b/java/java.lsp.server/test/unit/src/org/netbeans/modules/java/lsp/server/protocol/TextDocumentServiceImplTest.java
+@@ -18,14 +18,19 @@
+  */
+ package org.netbeans.modules.java.lsp.server.protocol;
+ 
++import java.util.Collections;
++import java.util.List;
+ import java.util.concurrent.atomic.AtomicInteger;
+ import javax.swing.event.DocumentEvent;
+ import javax.swing.event.DocumentListener;
+ import javax.swing.text.BadLocationException;
+ import javax.swing.text.Document;
+ import javax.swing.text.PlainDocument;
++import org.eclipse.lsp4j.FoldingRange;
+ import org.netbeans.junit.NbTestCase;
+ 
++import static org.netbeans.modules.java.lsp.server.protocol.TextDocumentServiceImpl.convertToLineOnlyFolds;
++
+ public class TextDocumentServiceImplTest extends NbTestCase {
+ 
+     public TextDocumentServiceImplTest(String name) {
+@@ -117,4 +122,141 @@ public class TextDocumentServiceImplTest extends NbTestCase {
+             fail(String.valueOf(e));
+         }
+     }
++    
++    public void testConvertToLineOnlyFolds() {
++        assertNull(convertToLineOnlyFolds(null));
++        assertEquals(0, convertToLineOnlyFolds(Collections.emptyList()).size());
++        List<FoldingRange> inputFolds, outputFolds;
++        inputFolds = Collections.singletonList(createRange(10, 20));
++        assertEquals(inputFolds, convertToLineOnlyFolds(inputFolds));
++
++        // test stable sort by start index
++        inputFolds = List.of(createRange(10, 20, 9, 9), createRange(5, 9, 9, 9), createRange(10, 19, 9, 9), createRange(10, 14, 13, 13));
++        outputFolds = List.of(createRange(5, 9), createRange(10, 20), createRange(10, 19), createRange(10, 14));
++        assertEquals(outputFolds, convertToLineOnlyFolds(inputFolds));
++
++        // test already disjoint folds
++        inputFolds = List.of(createRange(10, 20, 9, 9), createRange(5, 9, 9, 9), createRange(15, 19, 13, 13), createRange(10, 14, 13, 13));
++        outputFolds = List.of(createRange(5, 9), createRange(10, 20), createRange(10, 14), createRange(15, 19));
++        assertEquals(outputFolds, convertToLineOnlyFolds(inputFolds));
++
++        // test invariant of range.endLine: there exists no otherRange.startLine == range.endLine.
++        inputFolds = List.of(createRange(10, 20, 35, 9), createRange(5, 10, 12, 9), createRange(15, 19, 20, 13), createRange(10, 15, 51, 13));
++        assertEquals(outputFolds, convertToLineOnlyFolds(inputFolds));
++
++        // test a complex example of a full file:
```

Comment:
Can we add this test case as automated in any manner, then it would be great?

Reply:
I'm not sure I understood the question. 

This is a test in the NetBeans test suite, and it'll execute in our automated netbeans test run.